src/critics/prefer_forward_critic.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 FastSense, Samsung Research
+
+#include "mppic/critics/prefer_forward_critic.hpp"
+
+namespace mppi::critics
+{
+
+void PreferForwardCritic::initialize()
+{
+  auto node = parent_.lock();
+
+  auto getParam = utils::getParamGetter(node, name_);
+  getParam(power_, "prefer_forward_cost_power", 1);
+  getParam(weight_, "prefer_forward_cost_weight", 10.0);
+
+  RCLCPP_INFO(
+    logger_, "PreferForwardCritic instantiated with %d power and %f weight.", power_, weight_);
+}
+
+void PreferForwardCritic::score(
+  const geometry_msgs::msg::PoseStamped & /* robot_pos */, const xt::xtensor<double, 3> & trajectories,
```

Comment:
Wouldn't the robot pose be useful to get the dx and dy's (since all would be in map frame). So subtracting the trajectories from the robot pose in tensor form give you those dx / dys?

Reply:
Seems more understandable than `xt::view(trajectories, xt::all(), xt::range(_, -1), 0 or 1);`

... of which I read as the last pose? I'm not sure why that would be what we'd be normalizing relative to (if `-1` means the same here as in python/numpy)

Reply:
maybe this isn't relevant

Reply:
robot pose is just a pose at time step 0, but we need angle between all time_steps angles and direction of translation